### ISSUE TITLE
Using "${foo:+word}" warns about foo being unasigned (SC2154)

### ISSUE THREAD

**Issue by sheeit** [koalaman/shellcheck#906]
- Rule Id: [SC2154](https://github.com/koalaman/shellcheck/wiki/SC2154).
- My shellcheck version: "online."
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/sh

# This scripts prints whether the variable $test is set or not.
test_is_set="${test:+set}";
test_is_set="${test_is_set:-unset or null}";

printf '"%stest" is %s\n' '$' "${test_is_set}";
```

#### Here's what shellcheck currently says:

    $ shellcheck myscript
    Line 4:
    test_is_set="${test:+set}";
                 ^-- SC2154: test is referenced but not assigned (for output from commands, use "$(test ...)" ).
    

#### Here's what I wanted or expected to see:

    $ shellcheck myscript
    No issues detected!
    

According to the [POSIX shell specification](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02), `${test:+word}` should evaluate to 'word' if $test is null or *unset* (i.e., not assigned); and therefore shellcheck should not warn about it.

Here's a test for the script above:

```sh
$ test='' sh test_is_set.sh 
"$test" is unset or null
$ unset test; sh test_is_set.sh 
"$test" is unset or null
$ test=foo sh test_is_set.sh 
"$test" is set
$
